e2e/runner/solana.go

```diff
@@ -0,0 +1,84 @@
+package runner
+
+import (
+	"context"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/stretchr/testify/require"
+
+	solanacontract "github.com/zeta-chain/zetacore/zetaclient/chains/solana/contract"
+)
+
+// GatewayProgramID is the program ID for the gateway program
+func (r *E2ERunner) GatewayProgramID() solana.PublicKey {
+	return solana.MustPublicKeyFromBase58(solanacontract.GatewayProgramID)
+}
+
+// ComputePdaAddress computes the PDA address for the gateway program
+func (r *E2ERunner) ComputePdaAddress() solana.PublicKey {
+	seed := []byte(solanacontract.PDASeed)
+	GatewayProgramID := solana.MustPublicKeyFromBase58(solanacontract.GatewayProgramID)
+	pdaComputed, bump, err := solana.FindProgramAddress([][]byte{seed}, GatewayProgramID)
+	require.NoError(r, err)
+
+	r.Logger.Info("computed pda: %s, bump %d\n", pdaComputed, bump)
+
+	return pdaComputed
+}
+
+// CreateSignedTransaction creates a signed transaction from instructions
+func (r *E2ERunner) CreateSignedTransaction(
+	instructions []solana.Instruction,
+	privateKey solana.PrivateKey,
+) *solana.Transaction {
+	// get a recent blockhash
+	recent, err := r.SolanaClient.GetRecentBlockhash(context.TODO(), rpc.CommitmentFinalized)
```

Comment:
Can we use runner context instead of TODO

Reply:
https://github.com/zeta-chain/node/pull/2465/commits/eb814a91d89bee5be48174081af7338f76cb197a